include/aspect/geometry_model/two_merged_boxes.h

```diff
@@ -203,6 +228,11 @@ namespace aspect
          */
         bool use_merged_grids;
 
+        /**
+         * A pointer to the initial topography model.
+         */
+        InitialTopographyModel::Interface<dim> *topo_model;
+
```

Comment:
I dont think you need to store this pointer. See my PR #6552 for how to do this without storing the pointer, in particular [here](https://github.com/geodynamics/aspect/pull/6552/files#diff-c6f6b62cf2f45de862e0151446daf34cd446c80f37b562db84323f4ef0ba1de9L42).

Reply:
I see, thank you for sending the details on implementation!